seahorn/jobs2/byte_buf_write_be322/aws_byte_buf_write_be32_harness2.c

```diff
@@ -17,17 +17,18 @@ int main() {
     assume(aws_byte_buf_is_valid(&buf));
 
     /* save current state of the parameters */
-    struct aws_byte_buf old = buf;
-    struct store_byte_from_buffer old_byte_from_buf;
-    save_byte_from_array(buf.buffer, buf.len, &old_byte_from_buf);
+    size_t old_len = buf.len;
+    size_t old_capacity = buf.capacity;
+    sea_tracking_on();
 
     /* operation under verification */
     if (aws_byte_buf_write_be32(&buf, x)) {
-        sassert(buf.len == old.len + 4);
-        sassert(old.capacity == buf.capacity);
-        sassert(old.allocator == buf.allocator);
+        sassert(buf.len == old_len + 4);
+        sassert(buf.capacity == old_capacity);
+        sassert(!sea_is_modified((char *)buf.allocator));
     } else {
-        assert_byte_buf_equivalence(&buf, &old, &old_byte_from_buf);
+        sassert(!sea_is_modified((char *)&buf));
+        sassert(!sea_is_modified((char *)buf.buffer));
```

Comment:
Whenever checking `assert_byte_buf_equivalence`, let's check that pointed to memory is also unmodified i.e. memory pointed to by `allocator` is unmodified.

https://sourcegraph.com/github.com/seahorn/verify-c-common@master/-/blob/seahorn/lib/utils.c?L74&subtree=true

Reply:
I have added `allocator` checks to all `assert_byte_buf_equivalence` replacements.